src/fromager/overrides.py

```diff
@@ -128,7 +128,12 @@ def find_override_method(distname: str, method: str) -> typing.Callable:
     return getattr(mod, method)
 
 
-def list_all(patches_dir: pathlib.Path, envs_dir: pathlib.Path, test: bool = False):
+def list_all(
+    patches_dir: pathlib.Path,
+    envs_dir: pathlib.Path,
+    download_source: dict[str, dict[str, str]],
```

Comment:
@dhellmann couldn't pass the settings object directly here since that was resulting in a circular import

Reply:
OK, maybe that's why all of those other arguments are separate, too.

For the linter in our downstream repo we'll need to figure out how to get that code to read the fromager settings and then pass the source, but that's not a big deal.